shared/js/toc.jsx

```diff
@@ -225,24 +224,32 @@ loop.shared.toc = (function(mozL10n) {
       }
 
       // XXX akita: Code below is just for testing purpose
+      var urlObj;
       var btn = this.refs.addSiteBtn;
       btn.textContent = "Loading...";
       btn.disabled = true;
 
-      loop.shared.utils.getPageMetadata(url).then(result => {
+      // Create an URL object in order to use a favicon until the actual
+      // metadata and thumbnail is ready.
+      try {
+        urlObj = new URL(url);
+      } catch (e) {
         this.props.dispatcher.dispatch(new sharedActions.ShowSnackbar({
-          label: mozL10n.get("snackbar_page_added")
+          label: mozL10n.get("snackbar_page_not_added")
         }));
```

Comment:
We can still showing the snackbar. Just dispatch the action from the `AddedPage` method